tools/packaging/packaging.go

```diff
@@ -103,48 +109,97 @@ func CreateLinuxPackages(osqueryVersion, hostname, secret string, insecure, inse
 		}
 	}
 
-	// Create the systemd unit file for the launcher service
-	systemdPath := filepath.Join(systemdDirectory, "launcher.service")
-	systemdFile, err := os.Create(filepath.Join(packageRoot, systemdPath))
-	if err != nil {
-		return "", "", errors.Wrap(err, "could not create launcher systemd unit file")
-	}
-	defer systemdFile.Close()
+	if systemd {
+		// Create the systemd unit file for the launcher service
+		systemdPath := filepath.Join(serviceDirectory, "launcher.service")
+		systemdFile, err := os.Create(filepath.Join(packageRoot, systemdPath))
+		if err != nil {
+			return "", "", errors.Wrap(err, "could not create launcher systemd unit file")
+		}
+		defer systemdFile.Close()
 
-	if updateChannel == "" {
-		updateChannel = "stable"
-	}
+		if updateChannel == "" {
+			updateChannel = "stable"
+		}
 
-	opts := &systemdTemplateOptions{
-		ServerHostname: grpcServerForHostname(hostname),
-		RootDirectory:  rootDirectory,
-		SecretPath:     secretPath,
-		OsquerydPath:   filepath.Join(binaryDirectory, "osqueryd"),
-		LauncherPath:   filepath.Join(binaryDirectory, "launcher"),
-		Insecure:       insecure,
-		InsecureGrpc:   insecureGrpc,
-		Autoupdate:     autoupdate,
-		UpdateChannel:  updateChannel,
-	}
-	if err := renderSystemdService(systemdFile, opts); err != nil {
-		return "", "", errors.Wrap(err, "could not render systemd unit file")
-	}
+		opts := &systemdTemplateOptions{
```

Comment:
the struct for systemd and init looks almost identical. Is there a way they can be shared?